pasystem/pasystem-impl/impl/src/java/org/sakaiproject/pasystem/impl/rest/PASystemEntityProvider.java

```diff
@@ -183,6 +183,11 @@ public String getTimezoneToolUrlForUser() {
             User thisUser = UserDirectoryService.getCurrentUser();
             String userid = thisUser.getId();
 
+            // If there is no user (e.g. on the gateway site!) there's no timezone
+            if(userid == null || userid == "") {
```

Comment:
Had a quick look, looks like it might return an anonymous user which means User isn't null. Strange pattern. In any case:
` if(userid == null || userid == "") {`
to
`if(StringUtils.isBlank(userid)) {`
👍